components/infobox/wikis/starcraft2/infobox_person_player.lua

```diff
@@ -0,0 +1,539 @@
+	end
+	return achievements
+end
+
+--kick the default history display
+--because we will add it a bit lower again
+function StarCraft2Player:getHistory() return nil end
+
+function StarCraft2Player:addCustomContent(infobox, args)
+	local retired
+	--only display retired if it contains a year
+	if string.match(args.retired or '', '%d%d%d%d%') then
+		retired = args.retired
+	end
+	local allkills
+	if shouldStoreData then
+		allkills = StarCraft2Player._getAllkills()
+		if allkills then
+			allkills = _ALLKILLICON .. allkills
+		end
+	end
+	infobox:header('Achievements', (not hasAchievements) and allkills or nil)
+	infobox:cell('All-kills', allkills)
+	infobox:header('History', args.history)
+	infobox:centeredCell(args.history)
+	infobox:cell('Retired', retired)
+
+	return infobox
+end
+
+function StarCraft2Player._getAllkills()
+	if shouldStoreData then
+		local allkillsData = mw.ext.LiquipediaDB.lpdb('datapoint', {
+			conditions = '[[pagename::' .. pagename .. ']] AND [[type::allkills]]',
+			query = 'information',
+			limit = 1
+		})
+		if type(allkillsData[1]) == 'table' then
+			return allkillsData[1].information
+		end
+	end
+end
+
+function StarCraft2Player:getRole(args)
+	local role = args.role or args.occupation or 'player'
+	role = string.lower(role)
+	local category = _ROLES[role]
+	local store = category or _CLEAN_OTHER_ROLES[role] or 'Player'
+
+	return { title = 'Race', display = raceData.display, store = store, category = category or 'Player'}
+end
+
+function StarCraft2Player:calculateEarnings(args)
+	shouldStoreData = Player.shouldStoreData(args)
+
+	if shouldStoreData then
+		local earningsTotal
+		earningsTotal, earningsGlobal = StarCraft2Player._getEarningsMedalsData(self.pagename)
+		earningsTotal = math.floor( (earningsTotal or 0) * 100 + 0.5) / 100
+		return earningsTotal
+	end
+	return 0
+end
+
+function StarCraft2Player._getLPDBrecursive(cond, query, queryType)
+	local data = {} -- get LPDB results in here
+	local count
+	local offset = 0
+	repeat
+		local additionalData = mw.ext.LiquipediaDB.lpdb(queryType, {
+			conditions = cond,
+			query = query,
+			offset = offset,
+			limit = 5000
+		})
+		count = #additionalData
+		-- Merging
+		for i, item in ipairs(additionalData) do
+			data[offset + i] = item
+		end
+		offset = offset + count
+	until count ~= 5000
+
+	return data
+end
+
+function StarCraft2Player._getEarningsMedalsData(player)
+	local cond = '[[date::!1970-01-01 00:00:00]] AND ([[prizemoney::>0]] OR ' ..
+		'([[mode::1v1]] AND ([[placement::1]] OR [[placement::2]] OR [[placement::3]]' ..
+		' OR [[placement::4]] OR [[placement::3-4]]))) AND ' ..
+		'[[participant::' .. player .. ']]'
+	local query = 'liquipediatier, liquipediatiertype, placement, date, prizemoney, mode'
+
+	local data = StarCraft2Player._getLPDBrecursive(cond, query, 'placement')
+
+	local earnings = {}
+	local Medals = {}
+	earnings['total'] = {}
+	Medals['total'] = {}
+	local earnings_total = 0
+
+	if type(data[1]) == 'table' then
```

Comment:
This Module is SC2 specific so I'm not going to be too strict, but this code is very dense. Optional whether you want to change this:

- `Medals` -> `medals`
- Extract the earnings calculation per iteration to a function, and do the same for the medals. So you get something like:
```lua
for _, value in pairs(placements) do
    self:addPlacementToEarnings(.......)
    self:addPlacementToMedals(.......)
end
```

Could also combine the tables?

Reply:
those functions are still very much a work in progress
the review request was mostly about the commons module^^